src/saveDocs.js

```diff
@@ -4,45 +4,62 @@ import { getDownloads, getDependents } from './npm.js';
 import { getChangelogs } from './changelog.js';
 import { getHits } from './jsDelivr.js';
 import { getTSSupport } from './typescriptSupport.js';
+import datadog from './datadog.js';
+
+export default async function saveDocs({ docs, index }) {
+  const start = Date.now();
 
-export default function saveDocs({ docs, index }) {
   const rawPkgs = docs
     .filter(result => result.doc.name !== undefined) // must be a document
-    .map(result => formatPkg(result.doc))
+    .map(result => {
+      const start1 = Date.now();
+
+      const formatted = formatPkg(result.doc);
+
+      datadog.timing('formatPkg', Date.now() - start1);
+      return formatted;
+    })
     .filter(pkg => pkg !== undefined);
 
   if (rawPkgs.length === 0) {
     log.info('🔍 No pkgs found in response.');
     return Promise.resolve();
   }
 
-  return addMetaData(rawPkgs)
-    .then(pkgs => index.saveObjects(pkgs))
-    .then(() => log.info('🔍 Found and saved %d packages', rawPkgs.length));
+  let start2 = Date.now();
+  const pkgs = await addMetaData(rawPkgs);
+  datadog.timing('saveDocs.addMetaData', Date.now() - start2);
+
+  start2 = Date.now();
+  index.saveObjects(pkgs);
+  datadog.timing('saveDocs.saveObjects', Date.now() - start2);
+
+  datadog.timing('saveDocs', Date.now() - start);
+  return pkgs.length;
 }
 
-function addMetaData(pkgs) {
-  return Promise.all([
+async function addMetaData(pkgs) {
+  const [downloads, dependents, changelogs, hits, ts] = await Promise.all([
     getDownloads(pkgs),
     getDependents(pkgs),
     getChangelogs(pkgs),
     getHits(pkgs),
     getTSSupport(pkgs),
-  ]).then(([downloads, dependents, changelogs, hits, ts]) =>
-    pkgs.map((pkg, index) => ({
-      ...pkg,
-      ...downloads[index],
-      ...dependents[index],
-      ...changelogs[index],
-      ...hits[index],
-      ...ts[index],
-      _searchInternal: {
-        ...pkg._searchInternal,
-        ...downloads[index]._searchInternal,
-        ...dependents[index]._searchInternal,
-        ...changelogs[index]._searchInternal,
-        ...hits[index]._searchInternal,
-      },
-    }))
-  );
+  ]);
+
+  return pkgs.map((pkg, index) => ({
```

Comment:
I think we should add some timings around the whole block here too. I wonder if it could be improved by partialUpdateObject or something like that, or less looping & spreading (constructing the object key by key might be faster?).